flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCOutputFormat.java

```diff
@@ -202,14 +202,20 @@ public void writeRecord(Row row) throws IOException {
 			upload.addBatch();
 			batchCount++;
 			if (batchCount >= batchInterval) {
-				upload.executeBatch();
-				batchCount = 0;
+				flush();
 			}
 		} catch (SQLException | IllegalArgumentException e) {
 			throw new IllegalArgumentException("writeRecord() failed", e);
 		}
 	}
 
+	void flush() throws SQLException {
+		if (upload != null) {
+			upload.executeBatch();
```

Comment:
It is a synchronous call. It will throw `SQLException` and abort the sink. The behavior has not been changed.